### PR TITLE
Enable setting configuration options for local testing

### PR DESCRIPTION
**Pull request by cnagadya**
Adds ability to set configuration options via [dev-config.yml](https://github.com/actions/dependency-review-action/blob/c84947f64ba1957059efc3610314bf72a5d59929/.github/dev-config.yml)

PS: With modification of [CONTRIBUTING.md](https://github.com/actions/dependency-review-action/blob/c84947f64ba1957059efc3610314bf72a5d59929/CONTRIBUTING.md) to update instructions, there are some linter changes included in this PR

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -56,16 +55,22 @@ Like this:
 $ GITHUB_TOKEN=my-secret-token ./scripts/scan_pr https://github.com/actions/dependency-review-action/pull/3
 ```
 
+[Configuration options](README.md#configuration-options) can be set via the [external config file](README.md#configuration-file) at [.github/dev-config](.github/dev-config.yml) and the script run the with a `--file` or `-f` flag
+
+```sh
+$ GITHUB_TOKEN=<token> ./scripts/scan_pr <pr_url> --file
```
Comment (febuiles) [inline CONTRIBUTING.md:61]: Should users be able to specify the filename instead of hardcoding it?

**Comment by febuiles** [comment]
Validation:

Screenshot 2022-10-21 at 14 05 05

With an external config file:

Screenshot 2022-10-21 at 14 09 07

**Comment by febuiles** [comment]
@cnagadya please take a look and let me know what you think!

File: scripts/scan_pr [hunk 2]
```diff
@@ -3,22 +3,52 @@ require 'json'
...
-arg = /(?<repo_nwo>[\w\-]+\/[\w\-]+)\/pull\/(?<pr_number>\d+)/.match(ARGV[0])
+op = OptionParser.new do |opts|
+  usage = <<EOF
+Run Dependency Review on a repository.
+
+\e[1mUsage:\e[22m
+  script/scan_pr [options] <pr_url>
+
+\e[1mExample:\e[22m
+  script/scan_pr https://github.com/actions/dependency-review-action/pull/294
```
Comment (cnagadya) [inline scripts/scan_pr:29]:
```suggestion
  scripts/scan_pr https://github.com/actions/dependency-review-action/pull/294
```

File: scripts/scan_pr [hunk 3]
```diff
@@ -3,22 +3,52 @@ require 'json'
...
   exit -1
 end
 
-arg = /(?<repo_nwo>[\w\-]+\/[\w\-]+)\/pull\/(?<pr_number>\d+)/.match(ARGV[0])
+op = OptionParser.new do |opts|
+  usage = <<EOF
+Run Dependency Review on a repository.
+
+\e[1mUsage:\e[22m
+  script/scan_pr [options] <pr_url>
```
Comment (cnagadya) [inline scripts/scan_pr:26]:
```suggestion
  scripts/scan_pr [options] <pr_url>
```

**Comment by cnagadya** [comment]
Looks great. Just two small (`script` vs `scripts`) typos in [scan_pr](https://github.com/actions/dependency-review-action/blob/80e573b784829956d8a133b4bdb66153f515d053/scripts/scan_pr)

**Pull request merged**